build-tools/packages/build-cli/src/library/dangerfile.ts

```diff
@@ -47,6 +62,37 @@ export async function dangerfile(): Promise<void> {
 	// there were actual changes to the bundle sizes.  In other cases, we don't post a
 	// message and danger will delete its previous message
 	if (result.comparison === undefined || !bundlesContainNoChanges(result.comparison)) {
+		// Check for bundle size regression
+		let sizeCheck = false;
+		if (result.comparison !== undefined) {
+			for (const bundle of result.comparison) {
+				const totalMetric = bundle.commonBundleMetrics[totalSizeMetricName];
+				const totalParsedSizeDiff =
+					totalMetric.compare.parsedSize - totalMetric.baseline.parsedSize;
+				if (totalParsedSizeDiff > 5120) {
+					sizeCheck = true;
+					break;
+				}
+			}
+		}
+
+		// Warn and add label to PR in case of bundle size regression
+		if (sizeCheck) {
+			warn("Bundle size regression detected -- please investigate before merging!");
+			// Add the label to the PR
+			try {
+				await Promise.resolve(
+					danger.github.utils.createOrAddLabel({
+						color: "ff0000",
+						description: "Significant bundle size regression (>5 KB)",
```

Comment:
Suppose we change the threshold later - how does this API react if we alter the description over time (e.g. does it update the label, create a new second label, fail)?

Reply:
Will look into this

Reply:
From reading code and documentation, createOrAddLabel first looks to see if the label name exists in the repo's [label list](https://github.com/microsoft/FluidFramework/labels). If the label name already exists it just adds the existing label to the PR (i.e. no updated description). I think I'll separate the labeling functionality to it's own PR and use danger.github.api instead to have more custom control of the logic we want (updating existing label descriptions, removing label when regression is fixed, etc.). It'd be good to get the warning in to address the repair item and work on getting labeling right after.

Reply:
OK - to clarify, as long as it doesn't cause this tool to break in some way I think it's OK for a first go (we can always go manually change the description in the github UI if we make a policy change, folks can remove the label when they fix their own regression, etc.).  But also fine to start without the label if you prefer.